js/components/debug.tsx

```diff
@@ -81,171 +55,239 @@ class DebugViewer extends React.Component<IComponentProps, IDebugViewerState> {
-
-        const paths = this.state.paths;
+        const { paths } = this.state;
         const valueElementId = e.currentTarget.getAttribute('data-value-id');
         const pathPart = e.currentTarget.getAttribute('data-path-part');
 
         paths[valueElementId] =
-            ((paths[valueElementId] || '') + '.' + pathPart).replace(/^\./gi, '');
+            // `${(paths[valueElementId] || '')}.${pathPart})`.replace(/^\./gi, '');
+            (`${(paths[valueElementId] || '')}.${pathPart}`).replace(/^\./gi, '');
 
         this.setState({
             paths,
         });
     }
 
-    renderValues(title, id, values = [], name, idName) {
+    getValueByPath = (obj: any, path: string): any => {
+        let objectData = obj;
 
-        const isExpanded = this.state.toggle[id] || false;
+        if (!path || path === '') {
+            return objectData;
+        }
 
-        return <div className={'debug-root'} key={id}>
-            <div className={'debug-root-toggle'} id={id} onClick={this.toggleHeader}>
-                <span className={
-                    'glyphicon glyphicon-triangle-' + ((isExpanded) ? 'bottom' : 'right')
-                } />
-                <h5 className={'debug-title'}>
-                    {title}
-                </h5>
-                <span className={'label label-info'}>
-                    {values.length}
-                </span>
-            </div>
-            <ul className={'list-unstyled debug-values ' + ((isExpanded) ? '' : 'hidden')}>
-                {
-                    values.map(
-                        value => this.renderValue(
-                            this.state.paths[value[idName]] || '', value, name, idName,
-                        ),
-                        this,
-                    )
+        try {
+            const parts = path.split('.');
+
+            for (let i = 0; i < parts.length; i += 1) {
+                let foundKey = null;
+
+                Object.keys(objectData).forEach((key) => {
+                    if (key.toLowerCase() === parts[i].toLowerCase()) {
+                        foundKey = key;
+                    }
+                });
+
+                if (foundKey) {
+                    objectData = obj[foundKey];
+                }
+                else {
+                    objectData = undefined;
                 }
-            </ul>
-        </div>;
+            }
+            return objectData;
 
+        } catch (ex) {
+            return undefined;
+        }
+    };
+
+    toggleValue(e) {
+        e.stopPropagation();
+
+        const { toggle } = this.state;
+        const valueElementId = e.currentTarget.getAttribute('data-value-id');
+
+        toggle[valueElementId] = !toggle[valueElementId];
+
+        this.setState({
+            toggle,
+        });
     }
 
-    renderValue(path, value, name, idName) {
+    toggleHeader(e) {
+        const { toggle } = this.state;
+        toggle[e.currentTarget.id] = !toggle[e.currentTarget.id];
```

Comment:
Should this use dataset like the other one?

Reply:
Don't think so, as the element is not using a data attribute

Reply:
But shouldn't it? To be consistent with the others, rather than using an id attribute.